hybrid/pkg/server/server_test.go

```diff
@@ -0,0 +1,638 @@
+	combinedPayloads := []byte("a")
+	stream := StreamMock{CombinedPayloads: &combinedPayloads}
+	stream.returnError = nil
+	stream.Response = &nodeattestorv1.AttestResponse{
+		Response: &nodeattestorv1.AttestResponse_AgentAttributes{
+			AgentAttributes: &nodeattestorv1.AgentAttributes{
+				SelectorValues: []string{"test", "test2"},
+				SpiffeId:       "spiffe://example.org/spire/agent/k8s_psat/test",
+			},
+		},
+	}
+	interceptor.setCustomStream(&stream)
+	interceptor.spiffeid = ""
+	interceptorList := []ServerInterceptorInterface{}
+	interceptorList = append(interceptorList, interceptor.SpawnInterceptor())
+	interceptorList = append(interceptorList, interceptor.SpawnInterceptor())
+	interceptorList[0].(*InterceptorWrapper).canReattest = []bool{true}
+	interceptorList[0].(*InterceptorWrapper).spiffeid = "spiffe://example.org/spire/agent/k8s_psat/test"
+	interceptorList[1].(*InterceptorWrapper).canReattest = []bool{false}
+	interceptorList[1].(*InterceptorWrapper).spiffeid = "spiffe://example.org/spire/agent/aws_iid/test"
+	err := plugin.SendResponse(interceptorList)
+	require.NoError(t, err, "SendResponse failed: %v", err)
+	require.Equal(t, "spiffe://example.org/spire/agent/k8s_psat/test", interceptor.spiffeid, "Main interceptor spiffeid was not set")
+
+	fakeStore.SetAgentInfo(&agentstorev1.AgentInfo{
+		AgentId: "spiffe://example.org/spire/agent/aws_iid/test",
+	})
+	err = plugin.SendResponse(interceptorList)
+	require.EqualError(t, err, "rpc error: code = PermissionDenied desc = attestation data has already been used to attest an agent")
+
+	fakeStore.SetAgentErr("spiffe://example.org/spire/agent/aws_iid/test", status.Error(codes.InvalidArgument, "Error retrieving agentInfo"))
+	err = plugin.SendResponse(interceptorList)
+	require.EqualError(t, err, "rpc error: code = InvalidArgument desc = unable to get agent info: Error retrieving agentInfo", "SendResponse failed unexpectedly: %v", err)
+
+}
+
+func TestNew(t *testing.T) {
+	hybridPlugin := New()
+	require.NotNil(t, hybridPlugin, "New should return a non-nil value")
+	require.IsType(t, &HybridPluginServer{}, hybridPlugin, "New should return a HybridPluginServer")
+}
+
+func TestResetInterceptor(t *testing.T) {
+	req := nodeattestorv1.AttestRequest{Request: &nodeattestorv1.AttestRequest_Payload{Payload: nil}}
+	resp := nodeattestorv1.AttestResponse{Response: &nodeattestorv1.AttestResponse_AgentAttributes{
+		AgentAttributes: &nodeattestorv1.AgentAttributes{
+			SelectorValues: []string{"test", "test2"},
+			SpiffeId:       "spiffe://example.org/spire/agent/k8s_psat/test",
+			CanReattest:    false,
+		},
+	}}
+	interceptor := HybridPluginServerInterceptor{ctx: context.Background(), stream: StreamMock{}, logger: hclog.Default(), req: &req, Response: &resp,
+		combinedSelectors: []string{"test"}, spiffeID: "spiffe", canReattest: []bool{true}}
+
+	if interceptor.ctx == nil && interceptor.stream == nil && interceptor.logger == nil && interceptor.req == nil && interceptor.Response == nil &&
```

Comment:
What do you think the check structure is nil?
` if (interceptor == HybridPluginServerInterceptor{}) {
        errors.New("Interceptor is empty")
    }`